### DISCUSSION TITLE
How add my own custom settings.json to code-server ?

### DISCUSSION THREAD

**Discussion by niteshk9894** [coder/code-server#5974] [Q&A]
Can anyone let me know how can i add my own settings.json to code-server so that when vscode comesup it takes my custom settings.json file.

sample:
{
	// Controls whether the editor shows CodeLens.
	"diffEditor.codeLens": false,

	// 
	//  - smart: Uses the default diffing algorithm.
	//  - experimental: Uses an experimental diffing algorithm.
	"diffEditor.diffAlgorithm": "smart",

	// When enabled, the diff editor ignores changes in leading or trailing whitespace.
	"diffEditor.ignoreTrimWhitespace": true,
}

Thanks in advance.

**Comment by lg-nineoneninethreenine** [comment]
Hmm... Just change the settings file manually? Hit F1, type settings json, go to the file, input and save?

**Reply by jsjoeio** [reply]
You can also open the Command Palette > Open Settings (JSON) and add there

**Reply by niteshk9894** [reply]
i meant when code-sever can i give my custom settings.json ..... Does code-server provide some way for it. Not when vscode comesup and make changes in settings.json through Command Palette.

**Reply by jsjoeio** [reply]
Ah...so when code-server starts up, it will print a line about the `user-data-dir`. For instance I'm on macOS and it says this:
```shell
Using user-data-dir ~/.local/share/code-server
```
That's where it stores your `settings.json`

<img width="917" alt="image" src="https://user-images.githubusercontent.com/3806031/213261846-a8154bfa-3689-4bdd-b1b6-0db34ba2c578.png">

So you can either pass the `--user-data-dir` as a CLI flag and point to a new directory, or move your custom `settings.json` into the default location.

**Reply by niteshk9894** [reply]
this works thanks for the help. :)
